app/javascript/contentDisplayPolicy/initHiddenComments.js

```diff
@@ -42,20 +16,67 @@ For further actions, you may consider blocking this person and/or reporting abus
       });
   }
 
-  const hideLinks = Array.from(document.getElementsByClassName('hide-comment'));
+  function showHideCommentsModal(commentId) {
+    const form = document.getElementById('hide-comments-modal__form');
+    form.action = `/comments/${commentId}/hide`;
 
-  hideLinks.forEach((link) => {
-    const { hideType, commentId } = link.dataset;
+    window.Forem.showModal({
+      title: 'Confirm hiding the comment',
+      contentSelector: '#hide-comments-modal',
+      overlay: true,
+    }).then(() => {
+      const hideCommentForm = document.querySelector(
+        '#window-modal .hide-comments-modal__form',
+      );
 
-    if (hideType === 'hide') {
-      link.addEventListener('click', () => {
-        hide(commentId);
-      });
-    } else if (hideType === 'unhide') {
-      link.addEventListener('click', () => {
-        unhide(commentId);
-      });
-    }
+      hideCommentForm.addEventListener('submit', handleHideCommentsFormSubmit);
+    });
+  }
+
+  const hideButtons = Array.from(
+    document.getElementsByClassName('hide-comment'),
+  );
+
+  hideButtons.forEach((butt) => {
+    const { commentId } = butt.dataset;
+    butt.addEventListener('click', (e) => {
+      e.preventDefault();
+      showHideCommentsModal(commentId);
+    });
+  });
+
+  const unhideLinks = Array.from(
+    document.getElementsByClassName('unhide-comment'),
+  );
+
+  unhideLinks.forEach((link) => {
+    const { commentId } = link.dataset;
+    link.addEventListener('click', (e) => {
+      e.preventDefault();
+      unhide(commentId);
+    });
   });
+
+  const handleHideCommentsFormSubmit = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    const { target: form } = e;
+    const hide_children_check = form.getElementsByClassName('hide_children')[0];
+const url = `${form.action}${hide_children_check.checked ? '?hide_children=1' : ''}`
```

Comment:
(nitpick / not blocking) - Looks like a little bit of formatting has gone astray here